anta/cli/get/utils.py

```diff
@@ -204,3 +211,49 @@ def create_inventory_from_ansible(inventory: Path, output: Path, ansible_group:
         raise ValueError(msg)
     ansible_hosts = deep_yaml_parsing(ansible_inventory)
     write_inventory_to_file(ansible_hosts, output)
+
+
+def explore_package(module_name: str, level: int = 0, test_name: str | None = None, *, short: bool = False) -> None:
+    """Parse submodules recursively and print AntaTest example."""
+    if (module_spec := importlib.util.find_spec(module_name)) is None or module_spec.origin is None:
+        return
+
+    if module_spec.submodule_search_locations:
+        for _, sub_module_name, ispkg in pkgutil.walk_packages(module_spec.submodule_search_locations):
+            qname = f"{module_name}.{sub_module_name}"
+            if ispkg:
+                explore_package(qname, level=level + 1, test_name=test_name)
+                continue
+            print_tests_examples(qname, level, test_name, short=short)
+
+    else:
+        print_tests_examples(module_spec.name, level, test_name, short=short)
+
+
+def print_tests_examples(qname: str, level: int, test_name: str | None, *, short: bool = False) -> None:
+    """Print tests in qname if matching test_name (or all if test_name is None.
+
+    TODO: Allow to give a package argument to import_module.
+    """
+    qname_module = importlib.import_module(qname)
+    module_printed = False
+    for _name, obj in inspect.getmembers(qname_module):
+        if not inspect.isclass(obj) or not issubclass(obj, AntaTest) or obj == AntaTest:
+            continue
+        if test_name and not obj.name.startswith(test_name):
+            continue
+        if not module_printed:
+            console.print(f"{qname}:")
+            module_printed = True
+        console.print(f"  - {obj.name}:")
+        console.print(f"      # {obj.description}", soft_wrap=True)
+        if short:
+            continue
+        doc = NumpyDocString(obj.__doc__)
+        if "Examples" not in doc or not doc["Examples"]:
+            msg = f"Test {obj.name} in module {qname} is missing an Example"
+            raise LookupError(msg)
+        # Picking up only the inputs in the examples
+        if len(doc["Examples"]) > 4 + level:  # otherwise it is a test with no params.
+            inputs = "\n".join(doc["Examples"][level + 3 : -1])
+            console.print(textwrap.indent(textwrap.dedent(inputs), " " * 6))
```

Comment:
I think we should put these magic numbers in constants to understand what they represent and we can change them more easily in the future if needed.

Reply:
there is only one left which is the indent so I think this is addressed